src/mlpack/tests/mlpack_test.cpp

```diff
@@ -47,6 +106,22 @@ struct GlobalFixture
       mlpack::Log::Info.ignoreInput = true;
       mlpack::Log::Warn.ignoreInput = true;
     #endif
+
+    for (int i = 0; i < boost::unit_test::framework::master_test_suite().argc;
+        i++)
+    {
+      std::string argument(
+          boost::unit_test::framework::master_test_suite().argv[i]);
+
+      // Print Boost test hierarchy.
+      if (argument == "tests")
```

Comment:
Do you think we should name this `--print_tests` instead, so it is more like the other Boost UTF command line arguments?

I tried also catching `--help` to see if I could add documentation for `--print-tests`, but Boost UTF terminates the program after printing the help options, so I couldn't add documentation in that way.